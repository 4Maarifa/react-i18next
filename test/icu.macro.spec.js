import pluginTester from 'babel-plugin-tester';
import plugin from 'babel-plugin-macros';

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename, parserOpts: { plugins: ['jsx'] } },
  tests: [
    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Welcome, { name }!</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Welcome, <strong>{ name }</strong>!</Trans>
    `,

    `
      import { Trans } from '../icu.macro'
      import { useTranslation } from 'react-i18next'

      const x = <Trans>Trainers: { trainersCount, number }</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Trainers: <strong>{ trainersCount, number }</strong>!</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Caught on { catchDate, date, short }</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Caught on <strong>{ catchDate, date, short }</strong>!</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans defaults="Trainers: { trainersCount, number }" />
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans i18nKey="trainersWithDefaults" defaults="Trainers: <strong>{ trainersCount, number }</strong>!" />
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans i18nKey="caughtWithDefaults" defaults="Caught on { catchDate, date, short }" />
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans defaults="Caught on <strong>{ catchDate, date, short }</strong>!" />
    `,

    `
      import { Trans } from '../icu.macro'
      const Link = ({to, children}) => (<a href={to}>{children}</a>)
      
      const x = <Trans defaults="Caught on <Link to='/dest'>{ catchDate, date, short }</Link>!" values={{catchDate: Date.now()}}>This should be overridden by defaults</Trans>
    `,

    `
      import { Trans } from '../icu.macro'
      
      const x = <Trans i18nKey="trainersWithDefaults" values={{trainersCount}} defaults="Trainers: <strong>{ trainersCount, number }</strong>!" components={[]} />
    `,

    `
      import { Select } from '../icu.macro'

      const x = <Select
        i18nKey="selectExample"
        switch={gender}
        male="He avoids bugs."
        female="She avoids bugs."
        other="They avoid bugs."
      />
    `,

    `
      import { Select } from '../icu.macro'

      const x = <Select
        switch={gender}
        male={<Trans><strong>He</strong> avoids bugs.</Trans>}
        female={<Trans><strong>She</strong> avoids bugs.</Trans>}
        other={<Trans><strong>They</strong> avoid bugs.</Trans>}
      />
    `,

    `
      import { Plural } from '../icu.macro'

      const x = <Plural
        count={itemsCount1}
        $0="There are no items."
        one="There is # item."
        other="There are # items."
      />
    `,

    `
      import { Plural } from '../icu.macro'

      const x = <Plural
        i18nKey="testKey"
        count={itemsCount3}
        $0={<Trans>There is <strong>no</strong> item.</Trans>}
        one={<Trans>There is <strong>#</strong> item.</Trans>}
        other={<Trans>There are <strong>#</strong> items.</Trans>}
      />
    `,

    `
      import { SelectOrdinal } from '../icu.macro'

      const x = <SelectOrdinal
        count={position}
        zero="You are #th in line"
        one="You are #st in line"
        two="You are #nd in line"
        few="You are #rd in line"
        many="You are #th in line"
        other="You are #th in line"
        $7="You are in the lucky #th place in line"
      />
    `,

    `
      import { SelectOrdinal } from '../icu.macro'

      const x = <SelectOrdinal
        i18nKey="testKey"
        count={position}
        one={<Trans>You are <strong>#st in line</strong></Trans>}
        two={<Trans>You are <strong>#nd in line</strong></Trans>}
        few={<Trans>You are <strong>#rd in line</strong></Trans>}
        other={<Trans>You are <strong>#th in line</strong></Trans>}
        $0={<Trans>You are <strong>#th in line</strong></Trans>}
      />
    `,
  ],
});
