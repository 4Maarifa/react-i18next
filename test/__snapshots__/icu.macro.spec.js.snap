// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Welcome, { name }!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"Welcome, {name}!\\" components={[]} values={{
  name
}} />;
"
`;

exports[`macros 2. macros: 2. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Welcome, <strong>{ name }</strong>!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"Welcome, <0>{name}</0>!\\" components={[<strong>{name}</strong>]} values={{
  name
}} />;
"
`;

exports[`macros 3. macros: 3. macros 1`] = `
"
import { Trans } from '../icu.macro'
import { useTranslation } from 'react-i18next'

const x = <Trans>Trainers: { trainersCount, number }</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { useTranslation, Trans } from 'react-i18next';
const x = <Trans defaults=\\"Trainers: {trainersCount, number}\\" components={[]} values={{
  trainersCount
}} />;
"
`;

exports[`macros 4. macros: 4. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Trainers: <strong>{ trainersCount, number }</strong>!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"Trainers: <0>{trainersCount, number}</0>!\\" components={[<strong>{(trainersCount)}</strong>]} values={{
  trainersCount
}} />;
"
`;

exports[`macros 5. macros: 5. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Caught on { catchDate, date, short }</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"Caught on {catchDate, date, short}\\" components={[]} values={{
  catchDate
}} />;
"
`;

exports[`macros 6. macros: 6. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Caught on <strong>{ catchDate, date, short }</strong>!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"Caught on <0>{catchDate, date, short}</0>!\\" components={[<strong>{(catchDate)}</strong>]} values={{
  catchDate
}} />;
"
`;

exports[`macros 7. macros: 7. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans defaults=\\"Trainers: { trainersCount, number }\\" />

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"Trainers: {trainersCount, number}\\" components={[]} values={{
  trainersCount
}} />;
"
`;

exports[`macros 8. macros: 8. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans i18nKey=\\"trainersWithDefaults\\" defaults=\\"Trainers: <strong>{ trainersCount, number }</strong>!\\" />

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans i18nKey=\\"trainersWithDefaults\\" defaults=\\"Trainers: <0>{trainersCount, number}</0>!\\" components={[<strong>{(trainersCount)}</strong>]} values={{
  trainersCount
}} />;
"
`;

exports[`macros 9. macros: 9. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans i18nKey=\\"caughtWithDefaults\\" defaults=\\"Caught on { catchDate, date, short }\\" />

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans i18nKey=\\"caughtWithDefaults\\" defaults=\\"Caught on {catchDate, date, short}\\" components={[]} values={{
  catchDate
}} />;
"
`;

exports[`macros 10. macros: 10. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans defaults=\\"Caught on <strong>{ catchDate, date, short }</strong>!\\" />

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"Caught on <0>{catchDate, date, short}</0>!\\" components={[<strong>{(catchDate)}</strong>]} values={{
  catchDate
}} />;
"
`;

exports[`macros 11. macros: 11. macros 1`] = `
"
import { Trans } from '../icu.macro'
const Link = ({to, children}) => (<a href={to}>{children}</a>)

const x = <Trans defaults=\\"Caught on <Link to='/dest'>{ catchDate, date, short }</Link>!\\" values={{catchDate: Date.now()}}>This should be overridden by defaults</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";

const Link = ({
  to,
  children
}) => <a href={to}>{children}</a>;

const x = <Trans values={{
  catchDate: Date.now()
}} defaults=\\"Caught on <0>{catchDate, date, short}</0>!\\" components={[<Link to='/dest'>{(catchDate)}</Link>]} />;
"
`;

exports[`macros 12. macros: 12. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans i18nKey=\\"trainersWithDefaults\\" values={{trainersCount}} defaults=\\"Trainers: <strong>{ trainersCount, number }</strong>!\\" components={[]} />

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans i18nKey=\\"trainersWithDefaults\\" values={{
  trainersCount
}} defaults=\\"Trainers: <strong>{ trainersCount, number }</strong>!\\" components={[]} />;
"
`;

exports[`macros 13. macros: 13. macros 1`] = `
"
import { Select } from '../icu.macro'

const x = <Select
  i18nKey=\\"selectExample\\"
  switch={gender}
  male=\\"He avoids bugs.\\"
  female=\\"She avoids bugs.\\"
  other=\\"They avoid bugs.\\"
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans i18nKey=\\"selectExample\\" defaults=\\"{gender, select,  male {He avoids bugs.} female {She avoids bugs.} other {They avoid bugs.}}\\" components={[]} values={{
  gender
}} />;
"
`;

exports[`macros 14. macros: 14. macros 1`] = `
"
import { Select } from '../icu.macro'

const x = <Select
  switch={gender}
  male={<Trans><strong>He</strong> avoids bugs.</Trans>}
  female={<Trans><strong>She</strong> avoids bugs.</Trans>}
  other={<Trans><strong>They</strong> avoid bugs.</Trans>}
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"{gender, select,  male {<0>He</0> avoids bugs.} female {<1>She</1> avoids bugs.} other {<2>They</2> avoid bugs.}}\\" components={[<strong>He</strong>, <strong>She</strong>, <strong>They</strong>]} values={{
  gender
}} />;
"
`;

exports[`macros 15. macros: 15. macros 1`] = `
"
import { Plural } from '../icu.macro'

const x = <Plural
  count={itemsCount1}
  $0=\\"There are no items.\\"
  one=\\"There is # item.\\"
  other=\\"There are # items.\\"
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"{itemsCount1, plural,  =0 {There are no items.} one {There is # item.} other {There are # items.}}\\" components={[]} values={{
  itemsCount1
}} />;
"
`;

exports[`macros 16. macros: 16. macros 1`] = `
"
import { Plural } from '../icu.macro'

const x = <Plural
  i18nKey=\\"testKey\\"
  count={itemsCount3}
  $0={<Trans>There is <strong>no</strong> item.</Trans>}
  one={<Trans>There is <strong>#</strong> item.</Trans>}
  other={<Trans>There are <strong>#</strong> items.</Trans>}
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans i18nKey=\\"testKey\\" defaults=\\"{itemsCount3, plural,  =0 {There is <0>no</0> item.} one {There is <1>#</1> item.} other {There are <2>#</2> items.}}\\" components={[<strong>no</strong>, <strong>#</strong>, <strong>#</strong>]} values={{
  itemsCount3
}} />;
"
`;

exports[`macros 17. macros: 17. macros 1`] = `
"
import { SelectOrdinal } from '../icu.macro'

const x = <SelectOrdinal
  count={position}
  zero=\\"You are #th in line\\"
  one=\\"You are #st in line\\"
  two=\\"You are #nd in line\\"
  few=\\"You are #rd in line\\"
  many=\\"You are #th in line\\"
  other=\\"You are #th in line\\"
  $7=\\"You are in the lucky #th place in line\\"
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans defaults=\\"{position, selectordinal,  zero {You are #th in line} one {You are #st in line} two {You are #nd in line} few {You are #rd in line} many {You are #th in line} other {You are #th in line} =7 {You are in the lucky #th place in line}}\\" components={[]} values={{
  position
}} />;
"
`;

exports[`macros 18. macros: 18. macros 1`] = `
"
import { SelectOrdinal } from '../icu.macro'

const x = <SelectOrdinal
  i18nKey=\\"testKey\\"
  count={position}
  one={<Trans>You are <strong>#st in line</strong></Trans>}
  two={<Trans>You are <strong>#nd in line</strong></Trans>}
  few={<Trans>You are <strong>#rd in line</strong></Trans>}
  other={<Trans>You are <strong>#th in line</strong></Trans>}
  $0={<Trans>You are <strong>#th in line</strong></Trans>}
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react-i18next\\";
const x = <Trans i18nKey=\\"testKey\\" defaults=\\"{position, selectordinal,  one {You are <0>#st in line</0>} two {You are <1>#nd in line</1>} few {You are <2>#rd in line</2>} other {You are <3>#th in line</3>} =0 {You are <4>#th in line</4>}}\\" components={[<strong>#st in line</strong>, <strong>#nd in line</strong>, <strong>#rd in line</strong>, <strong>#th in line</strong>, <strong>#th in line</strong>]} values={{
  position
}} />;
"
`;
